### PR TITLE
Make GoogleStorageDAO.deleteBucket handle bucket not existing

### PR DESCRIPTION
**Pull request by rtitle**
Needed by Leo for pause/resume clusters.

**PR checklist**
- [ ] For each library you've modified here, decide whether it requires a major, minor, or no version bump. (Click [here](/broadinstitute/workbench-libs/blob/develop/CONTRIBUTING.md) for guidance)

If you're doing a **major** or **minor** version bump to any libraries:
- [ ] Bump the version in `project/Settings.scala` `createVersion()`
- [ ] Update `CHANGELOG.md` for those libraries
- [ ] I promise I used `@deprecated` instead of deleting code where possible

In all cases:
- [ ] Get two thumbsworth of PR review
- [ ] Verify all tests go green (CI _and_ coverage tests)
- [ ] Squash commits and **merge to develop**
- [ ] Delete branch after merge

After merging, _even if you haven't bumped the version_: 
- [ ] Update `README.md` and the `CHANGELOG.md` for any libs you changed with the new dependency string. The git hash is the same short version you see in GitHub (i.e. seven characters). You can commit these changes straight to develop.

### CODE REVIEW

File: google/src/main/scala/org/broadinstitute/dsde/workbench/google/HttpGoogleStorageDAO.scala [hunk 1]
```diff
@@ -65,9 +65,13 @@ class HttpGoogleStorageDAO(appName: String,
     // If `recurse` is true, first delete all objects in the bucket
     val deleteObjectsFuture = if (recurse) {
       val listObjectsRequest = storage.objects().list(bucketName.value)
-      retryWhen500orGoogleError(() => executeGoogleRequest(listObjectsRequest)).flatMap { objects =>
+      retryWithRecoverWhen500orGoogleError { () =>
+        Option(executeGoogleRequest(listObjectsRequest))
```
Comment (jmthibault79) [inline google/src/main/scala/org/broadinstitute/dsde/workbench/google/HttpGoogleStorageDAO.scala:69]: Isn't this going to swallow all exceptions?

Comment (rtitle) [reply google/src/main/scala/org/broadinstitute/dsde/workbench/google/HttpGoogleStorageDAO.scala:69]: It should just recover from 404s, making this `Future` a no-op. Other errors should result in a failed Future. We recover from 404s below in `deleteBucketRequest` as well.

I hit this in Leo when resuming a paused cluster -- it triggers this code to delete the init bucket, but the init bucket has already been deleted when the cluster started up the first time. So we don't want this method to fail on 404 errors.

Comment (jmthibault79) [reply google/src/main/scala/org/broadinstitute/dsde/workbench/google/HttpGoogleStorageDAO.scala:69]: I mean the `Option()` here.  The recover for 404 looks fine.

Comment (rtitle) [reply google/src/main/scala/org/broadinstitute/dsde/workbench/google/HttpGoogleStorageDAO.scala:69]: Oh sorry. `Option(_)` shouldn't swallow exceptions (unlike `Try`), but it should handle null responses. (I never know when Google APIs are going to return null)

Comment (jmthibault79) [reply google/src/main/scala/org/broadinstitute/dsde/workbench/google/HttpGoogleStorageDAO.scala:69]: Ah!  Did not realize that.  Looks good.

**Review by jmthibault79 — APPROVED** [review]

**Pull request merged**